File: src/MonitoringPlatform.Domain/Entities/Customer.cs
using MonitoringPlatform.Domain.Enums;

namespace MonitoringPlatform.Domain.Entities
{
    public class Customer
    {
        public Guid Id { get; set; }
        public Guid TenantId { get; set; }

        public string CompanyName { get; set; } = null!;
        public string ContactName { get; set; } = null!;
        public string ContactEmail { get; set; } = null!;
        public string? ContactPhone { get; set; }
        public CustomerType CustomerType { get; set; }
        public string? Industry { get; set; }
        public string? Notes { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public Tenant Tenant { get; set; } = null!;
        public ICollection<CustomerCloudProvider> CustomerCloudProviders { get; set; } = new List<CustomerCloudProvider>();
    }
}

File: frontend/src/types/customer.ts
export type CustomerType = 0 | 1 | 2 | 3;

export type CloudProviderSummary = {
  id: string;
  name: string;
  slug: string;
};

export type Customer = {
  id: string;
  companyName: string;
  contactName: string;
  contactEmail: string;
  contactPhone?: string;
  customerType: CustomerType;
  industry?: string;
  notes?: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  cloudProviders: CloudProviderSummary[];
};

export type CreateCustomerPayload = {
  companyName: string;
  contactName: string;
  contactEmail: string;
  contactPhone?: string;
  customerType: CustomerType;
  industry?: string;
  notes?: string;
  cloudProviderIds: string[];
};

export type UpdateCustomerPayload = {
  companyName: string;
  contactName: string;
  contactEmail: string;
  contactPhone?: string;
  customerType: CustomerType;
  industry?: string;
  notes?: string;
  isActive: boolean;
  cloudProviderIds: string[];
};
